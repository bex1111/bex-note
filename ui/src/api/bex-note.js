import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('http://localhost:1080/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('http://localhost:1080/api/authorization', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title }
    });
    return response.data;
}

export const saveNote = async (title,content) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/save', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title, content }
    });
    return response.data;
}


