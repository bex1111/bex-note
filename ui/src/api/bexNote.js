import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('/api/authorize', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: {title}
    });
    return response.data;
}

export const saveNote = async (title, content) => {
    const response = await axios.post('/api/internal/note/save', {
        data: {title, content}
    }, {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const getContent = async (title) => {
    const response = await axios.post('/api/internal/note/content', {
            data: {title}
        },
        {
            headers: {
                'x-auth-token': tokenStore.token
            }
        });
    return response.data;
}
