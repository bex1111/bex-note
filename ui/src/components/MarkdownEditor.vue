<template>
      <MdEditor v-model="text"
                language="en-Us"
                :toolbars="toolbar"
      />
  </template>

<script setup>
import {ref} from 'vue';
import {MdEditor} from 'md-editor-v3';
import 'md-editor-v3/lib/style.css';

const text = ref('Hello Editor!');
const toolbar = [
  'revoke',
  'next',
  '-',
  'bold',
  'underline',
  'italic',
  '-',
  'title',
  'strikeThrough',
  'sub',
  'sup',
  'quote',
  'unorderedList',
  'orderedList',
  'task',
  '-',
  'codeRow',
  'code',
  'link',
  'table',
  '-',
  'pageFullscreen',
  'fullscreen',
  'preview',
  'previewOnly'
];

</script>

<style scoped>

</style>
