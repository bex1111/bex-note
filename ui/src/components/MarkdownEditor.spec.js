import { mount } from '@vue/test-utils';
import { vi } from 'vitest';
import { nextTick } from 'vue';
import MarkdownEditor from './MarkdownEditor.vue';

vi.mock('../api/bexNote', () => ({
  getContent: vi.fn().mockResolvedValue({ content: 'mocked content' })
}));


import { getContent } from '../api/bexNote';

describe('MarkdownEditor.vue', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders correctly and matches snapshot', () => {
    const wrapper = mount(MarkdownEditor, {
      props: {
        title: 'test-title',
        modelValue: 'initial content'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('calls getContent and updates content when title prop changes', async () => {
    const wrapper = mount(MarkdownEditor, {
      props: {
        title: '',
        modelValue: 'initial content'
      }
    });
    await wrapper.setProps({ title: 'new-title' });
    await nextTick();


    expect(getContent).toHaveBeenCalledWith('new-title');
    expect(getContent).toHaveBeenCalledTimes(1);

    expect(wrapper.emitted('update:modelValue')).toBeTruthy();
    expect(wrapper.emitted('update:modelValue')[0]).toEqual(['mocked content']);
  });

  it('does not call getContent when title is empty', async () => {
    const wrapper = mount(MarkdownEditor, {
      props: {
        title: 'initial-title',
        modelValue: 'initial content'
      }
    });

    await wrapper.setProps({ title: '' });
    await nextTick();

    expect(getContent).not.toHaveBeenCalled();
  });
});
