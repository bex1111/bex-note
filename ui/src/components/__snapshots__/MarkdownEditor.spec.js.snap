// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarkdownEditor.vue > renders correctly and matches snapshot 1`] = `
"<div id=\\"md-editor-v-0\\" class=\\"md-editor\\">
  <div class=\\"md-editor-toolbar-wrapper\\" id=\\"md-editor-v-0-toolbar-wrapper\\">
    <div class=\\"md-editor-toolbar\\">
      <div class=\\"md-editor-toolbar-left\\"><button class=\\"md-editor-toolbar-item\\" title=\\"revoke\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-reply-icon lucide-reply md-editor-icon\\">
            <path d=\\"M20 18v-2a4 4 0 0 0-4-4H4\\"></path>
            <path d=\\"m9 17-5-5 5-5\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"undo revoke\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-forward-icon lucide-forward md-editor-icon\\">
            <path d=\\"m15 17 5-5-5-5\\"></path>
            <path d=\\"M4 18v-2a4 4 0 0 1 4-4h12\\"></path>
          </svg>
          <!---->
        </button>
        <div class=\\"md-editor-divider\\"></div><button class=\\"md-editor-toolbar-item\\" title=\\"bold\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-bold-icon lucide-bold md-editor-icon\\">
            <path d=\\"M6 12h9a4 4 0 0 1 0 8H7a1 1 0 0 1-1-1V5a1 1 0 0 1 1-1h7a4 4 0 0 1 0 8\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"underline\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-underline-icon lucide-underline md-editor-icon\\">
            <path d=\\"M6 4v6a6 6 0 0 0 12 0V4\\"></path>
            <line x1=\\"4\\" x2=\\"20\\" y1=\\"20\\" y2=\\"20\\"></line>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"italic\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-italic-icon lucide-italic md-editor-icon\\">
            <line x1=\\"19\\" x2=\\"10\\" y1=\\"4\\" y2=\\"4\\"></line>
            <line x1=\\"14\\" x2=\\"5\\" y1=\\"20\\" y2=\\"20\\"></line>
            <line x1=\\"15\\" x2=\\"9\\" y1=\\"4\\" y2=\\"20\\"></line>
          </svg>
          <!---->
        </button>
        <div class=\\"md-editor-divider\\"></div><button class=\\"md-editor-toolbar-item\\" title=\\"title\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-heading-icon lucide-heading md-editor-icon\\">
            <path d=\\"M6 12h12\\"></path>
            <path d=\\"M6 20V4\\"></path>
            <path d=\\"M18 20V4\\"></path>
          </svg>
          <!---->
        </button>
        <div class=\\"md-editor-dropdown md-editor-dropdown-hidden\\">
          <div class=\\"md-editor-dropdown-overlay\\">
            <ul class=\\"md-editor-menu\\" role=\\"menu\\">
              <li class=\\"md-editor-menu-item md-editor-menu-item-title\\" role=\\"menuitem\\" tabindex=\\"0\\">Lv1 Heading</li>
              <li class=\\"md-editor-menu-item md-editor-menu-item-title\\" role=\\"menuitem\\" tabindex=\\"0\\">Lv2 Heading</li>
              <li class=\\"md-editor-menu-item md-editor-menu-item-title\\" role=\\"menuitem\\" tabindex=\\"0\\">Lv3 Heading</li>
              <li class=\\"md-editor-menu-item md-editor-menu-item-title\\" role=\\"menuitem\\" tabindex=\\"0\\">Lv4 Heading</li>
              <li class=\\"md-editor-menu-item md-editor-menu-item-title\\" role=\\"menuitem\\" tabindex=\\"0\\">Lv5 Heading</li>
              <li class=\\"md-editor-menu-item md-editor-menu-item-title\\" role=\\"menuitem\\" tabindex=\\"0\\">Lv6 Heading</li>
            </ul>
          </div>
        </div><button class=\\"md-editor-toolbar-item\\" title=\\"strikeThrough\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-strikethrough-icon lucide-strikethrough md-editor-icon\\">
            <path d=\\"M16 4H9a3 3 0 0 0-2.83 4\\"></path>
            <path d=\\"M14 12a4 4 0 0 1 0 8H6\\"></path>
            <line x1=\\"4\\" x2=\\"20\\" y1=\\"12\\" y2=\\"12\\"></line>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"subscript\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-subscript-icon lucide-subscript md-editor-icon\\">
            <path d=\\"m4 5 8 8\\"></path>
            <path d=\\"m12 5-8 8\\"></path>
            <path d=\\"M20 19h-4c0-1.5.44-2 1.5-2.5S20 15.33 20 14c0-.47-.17-.93-.48-1.29a2.11 2.11 0 0 0-2.62-.44c-.42.24-.74.62-.9 1.07\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"superscript\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-superscript-icon lucide-superscript md-editor-icon\\">
            <path d=\\"m4 19 8-8\\"></path>
            <path d=\\"m12 19-8-8\\"></path>
            <path d=\\"M20 12h-4c0-1.5.442-2 1.5-2.5S20 8.334 20 7.002c0-.472-.17-.93-.484-1.29a2.105 2.105 0 0 0-2.617-.436c-.42.239-.738.614-.899 1.06\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"quote\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-quote-icon lucide-quote md-editor-icon\\">
            <path d=\\"M16 3a2 2 0 0 0-2 2v6a2 2 0 0 0 2 2 1 1 0 0 1 1 1v1a2 2 0 0 1-2 2 1 1 0 0 0-1 1v2a1 1 0 0 0 1 1 6 6 0 0 0 6-6V5a2 2 0 0 0-2-2z\\"></path>
            <path d=\\"M5 3a2 2 0 0 0-2 2v6a2 2 0 0 0 2 2 1 1 0 0 1 1 1v1a2 2 0 0 1-2 2 1 1 0 0 0-1 1v2a1 1 0 0 0 1 1 6 6 0 0 0 6-6V5a2 2 0 0 0-2-2z\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"unordered list\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-list-icon lucide-list md-editor-icon\\">
            <path d=\\"M3 5h.01\\"></path>
            <path d=\\"M3 12h.01\\"></path>
            <path d=\\"M3 19h.01\\"></path>
            <path d=\\"M8 5h13\\"></path>
            <path d=\\"M8 12h13\\"></path>
            <path d=\\"M8 19h13\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"ordered list\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-list-ordered-icon lucide-list-ordered md-editor-icon\\">
            <path d=\\"M11 5h10\\"></path>
            <path d=\\"M11 12h10\\"></path>
            <path d=\\"M11 19h10\\"></path>
            <path d=\\"M4 4h1v5\\"></path>
            <path d=\\"M4 9h2\\"></path>
            <path d=\\"M6.5 20H3.4c0-1 2.6-1.925 2.6-3.5a1.5 1.5 0 0 0-2.6-1.02\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"task list\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-list-todo-icon lucide-list-todo md-editor-icon\\">
            <path d=\\"M13 5h8\\"></path>
            <path d=\\"M13 12h8\\"></path>
            <path d=\\"M13 19h8\\"></path>
            <path d=\\"m3 17 2 2 4-4\\"></path>
            <rect x=\\"3\\" y=\\"4\\" width=\\"6\\" height=\\"6\\" rx=\\"1\\"></rect>
          </svg>
          <!---->
        </button>
        <div class=\\"md-editor-divider\\"></div><button class=\\"md-editor-toolbar-item\\" title=\\"inline code\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-code-icon lucide-code md-editor-icon\\">
            <path d=\\"m16 18 6-6-6-6\\"></path>
            <path d=\\"m8 6-6 6 6 6\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"block-level code\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-square-code-icon lucide-square-code md-editor-icon\\">
            <path d=\\"m10 9-3 3 3 3\\"></path>
            <path d=\\"m14 15 3-3-3-3\\"></path>
            <rect x=\\"3\\" y=\\"3\\" width=\\"18\\" height=\\"18\\" rx=\\"2\\"></rect>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"link\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-link-icon lucide-link md-editor-icon\\">
            <path d=\\"M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71\\"></path>
            <path d=\\"M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"table\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-table-icon lucide-table md-editor-icon\\">
            <path d=\\"M12 3v18\\"></path>
            <rect width=\\"18\\" height=\\"18\\" x=\\"3\\" y=\\"3\\" rx=\\"2\\"></rect>
            <path d=\\"M3 9h18\\"></path>
            <path d=\\"M3 15h18\\"></path>
          </svg>
          <!---->
        </button>
        <div class=\\"md-editor-dropdown md-editor-dropdown-hidden\\">
          <div class=\\"md-editor-dropdown-overlay\\">
            <div class=\\"md-editor-table-shape\\">
              <div class=\\"md-editor-table-shape-row\\">
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
              </div>
              <div class=\\"md-editor-table-shape-row\\">
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
              </div>
              <div class=\\"md-editor-table-shape-row\\">
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
              </div>
              <div class=\\"md-editor-table-shape-row\\">
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
                <div class=\\"md-editor-table-shape-col\\">
                  <div class=\\"md-editor-table-shape-col-default\\"></div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"md-editor-divider\\"></div><button class=\\"md-editor-toolbar-item\\" title=\\"fullscreen in page\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-maximize2-icon lucide-maximize-2 md-editor-icon\\">
            <path d=\\"M15 3h6v6\\"></path>
            <path d=\\"m21 3-7 7\\"></path>
            <path d=\\"m3 21 7-7\\"></path>
            <path d=\\"M9 21H3v-6\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"fullscreen\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-expand-icon lucide-expand md-editor-icon\\">
            <path d=\\"m15 15 6 6\\"></path>
            <path d=\\"m15 9 6-6\\"></path>
            <path d=\\"M21 16v5h-5\\"></path>
            <path d=\\"M21 8V3h-5\\"></path>
            <path d=\\"M3 16v5h5\\"></path>
            <path d=\\"m3 21 6-6\\"></path>
            <path d=\\"M3 8V3h5\\"></path>
            <path d=\\"M9 9 3 3\\"></path>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item md-editor-toolbar-active\\" title=\\"preview\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-eye-icon lucide-eye md-editor-icon\\">
            <path d=\\"M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0\\"></path>
            <circle cx=\\"12\\" cy=\\"12\\" r=\\"3\\"></circle>
          </svg>
          <!---->
        </button><button class=\\"md-editor-toolbar-item\\" title=\\"preview only\\" type=\\"button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"lucide md-editor-icon lucide-view-icon lucide-view md-editor-icon\\">
            <path d=\\"M21 17v2a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-2\\"></path>
            <path d=\\"M21 7V5a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v2\\"></path>
            <circle cx=\\"12\\" cy=\\"12\\" r=\\"1\\"></circle>
            <path d=\\"M18.944 12.33a1 1 0 0 0 0-.66 7.5 7.5 0 0 0-13.888 0 1 1 0 0 0 0 .66 7.5 7.5 0 0 0 13.888 0\\"></path>
          </svg>
          <!---->
        </button>
      </div>
      <div class=\\"md-editor-toolbar-right\\"></div>
    </div>
  </div>
  <div class=\\"md-editor-content\\">
    <div class=\\"md-editor-content-wrapper\\">
      <div class=\\"md-editor-custom-scrollbar\\" style=\\"width: 50%;\\">
        <div class=\\"md-editor-input-wrapper\\">
          <div class=\\"cm-editor ͼ1 ͼ3 ͼ4 ͼ15\\">
            <div class=\\"cm-announced\\" aria-live=\\"polite\\"></div>
            <div tabindex=\\"-1\\" class=\\"cm-scroller\\">
              <div style=\\"tab-size: 4;\\" spellcheck=\\"false\\" autocorrect=\\"off\\" autocapitalize=\\"off\\" writingsuggestions=\\"false\\" translate=\\"no\\" contenteditable=\\"true\\" class=\\"cm-content cm-lineWrapping\\" role=\\"textbox\\" aria-multiline=\\"true\\" data-language=\\"markdown\\">
                <div class=\\"cm-line\\">initial content</div>
              </div>
              <div class=\\"cm-layer cm-layer-above cm-cursorLayer\\" aria-hidden=\\"true\\" style=\\"z-index: 150; animation-duration: 1200ms;\\"></div>
              <div class=\\"cm-layer cm-selectionLayer\\" aria-hidden=\\"true\\" style=\\"z-index: -2;\\"></div>
            </div>
          </div>
        </div>
        <div class=\\"md-editor-custom-scrollbar__track\\">
          <div class=\\"md-editor-custom-scrollbar__thumb\\"></div>
        </div>
      </div>
      <div class=\\"md-editor-resize-operate\\" style=\\"left: 50%; display: initial;\\"></div>
      <div class=\\"md-editor-custom-scrollbar\\" style=\\"flex: 1 1 0%;\\">
        <div id=\\"md-editor-v-0-preview-wrapper\\" class=\\"md-editor-preview-wrapper\\">
          <div id=\\"md-editor-v-0-preview\\" class=\\"md-editor-preview default-theme md-editor-scrn\\">
            <p data-line=\\"0\\">initial content</p>
          </div>
        </div>
        <!---->
        <div class=\\"md-editor-custom-scrollbar__track\\">
          <div class=\\"md-editor-custom-scrollbar__thumb\\"></div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"md-editor-footer\\">
    <div class=\\"md-editor-footer-left\\">
      <div class=\\"md-editor-footer-item\\"><label class=\\"md-editor-footer-label\\">Character Count:</label><span>15</span></div>
    </div>
    <div class=\\"md-editor-footer-right\\">
      <!---->
      <div class=\\"md-editor-footer-item\\"><label class=\\"md-editor-footer-label\\">Scroll Auto</label>
        <div class=\\"md-editor-checkbox md-editor-checkbox-checked\\"></div>
      </div>
    </div>
  </div>
</div>"
`;
